server/src/main/java/org/apache/druid/server/coordinator/rules/LoadRule.java

```diff
@@ -144,16 +144,16 @@ private void assign(
   private static List<ServerHolder> getFilteredHolders(
       final String tier,
       final DruidCluster druidCluster,
-      final Predicate<ServerHolder> predicate
+      Predicate<ServerHolder> predicate
   )
   {
     final NavigableSet<ServerHolder> queue = druidCluster.getHistoricalsByTier(tier);
     if (queue == null) {
       log.makeAlert("Tier[%s] has no servers! Check your cluster configuration!", tier).emit();
       return Collections.emptyList();
     }
-
-    return queue.stream().filter(predicate).collect(Collectors.toList());
+    Predicate<ServerHolder> isGeneral = s -> !s.isInMaintenance();
```

Comment:
Why not `isAvailable` or something along those lines since the implication is that these servers are available to assign segments to, and to go along with your other variable rename suggestion in balancer logic

Reply:
IMHO since this is specifically around maintenance then keeping the boolean methods related to the maintenance.

That being said, I'm curious if this predicate is in the wrong place. Should it just be part of the passed in predicate, and leave no "extra" or surprising predicate logic in this method?

Reply:
`getFilteredHolders` is always called to find servers which can be a target location for a segment, I think it should never return maintenance servers.
Changed to `isNotInMaintenance` ✅